### PR TITLE
Add different render modes

### LINKED ISSUE #4: Add support for optional trailing slash in `/?` scripts

### LINKED ISSUE #9: Add several renders for `btldr-view`
Choosing render doesn't not just changes the appearance of the page, but also sets environment variables to correctly render pages.

- `--render tldr`: TlDr-like render
- `--render tldr-colorful`: TlDr-like render with highlighting placeholders in different colors
- `--render docopt`: DocOpt-like render
- `--render docopt-colorful`: DocOpt-like render with highlighting placeholders in different colors

Also add option to hide examples and hide placeholder descriptions. Note that at least one of both things must be present to make placeholders correctly rendered.

### LINKED ISSUE #11: New tag `Help: <string help flag name>` for generating help example on rendering

### LINKED ISSUE #12: New tag `Version: <string version flag name>` for generating version example on rendering

### PR DESCRIPTION
**Pull request by EmilySeville7cfg**
closes #4 
closes #9
closes #11
closes #12

## ToDo

- [x] TlDr-like modes
- [x] docopt-like modes
- [x] Unique CLIP mode

TlDr-like and DocOpt-like modes rely heavily on regex parsing, while CLIP does more real and more complex parsing, that's why escaping mnemonics works here.

## Screenshots

image

image

## Notes

Note that this tool is a just working prototype. The real render will be written later in Go with proper parsing and more capabilities. It's the pain in the ass to support such things those mix Bash, Sed and AWK, but yeah, it's a good way to break your brain and prove that you can do some crazy stuff. It's more like a challenge.

### CODE REVIEW

**Comment by EmilySeville7cfg** [comment]
@kbdharun, @los-romka, @navarroaxel, can u check TlDr-like mode? Usage examples:

```md
clip-view -r tldr [
clip-view -r tldr-colorful [
```

**Comment by kbdharun** [comment]
How do I test it locally? Can you show some samples?

**Comment by blueglyph** [comment]
I don't see any clip-view command.

**Comment by EmilySeville7cfg** [comment]
> How do I test it locally? Can you show some samples?

@kbdharun, clone [this](https://github.com/emilyseville7cfg-better-tldr/cli-pages) repo locally and run smth like this: `clip-view -r tldr /home/emilyseville7cfg/Documents/mine/other/cli-pages/common/[.clip`. *.clip is a page to render.

**Comment by EmilySeville7cfg** [comment]
@blueglyph, it is [here](https://github.com/emilyseville7cfg-better-tldr/prototypes/blob/feature/add-different-render-modes/clip-view/clip-view.sh).

File: clip-view/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# Render for Better TlDr pages
+# Render for Command Line Interface Pages pages
```
Comment (navarroaxel) [inline clip-view/README.md:1]:
```suggestion
# Render for Command Line Interface Pages
```

File: clip-view/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 
-Render for Better TlDr pages.
+Render for Command Line Interface Pages pages.
```
Comment (navarroaxel) [inline clip-view/README.md:3]:
```suggestion
Render for Command Line Interface Pages.
```

**Comment by blueglyph** [comment]
Oh, I was in a hurry and took the main branch apparently, that's why I couldn't see the script.

Here's what it looks like here, I'd say that blue is a bad colour choice because it will always be difficult to read. Same for bright red. Just my opinion, others may like it.

That's with Manjaro Cinnamon. Both commands seem to give the same result.

image

On Windows with MinGW the blue is brighter.

image

**Comment by EmilySeville7cfg** [comment]
> Here's what it looks like here, I'd say that blue is a bad colour choice because it will always be difficult to read. Same for bright red. Just my opinion, others may like it.

Yay, it's hard to read definitely (for me too). I think the solution can be a theme support, to colorize everything at once as it takes some time to setup all environment variables to customize colors. :)

**Comment by blueglyph** [comment]
Yeah, it's not obvious with terminal colours. There are only a few possibilities and it's subjective. These days there are about as many people using a light theme than a dark theme.

Shells often have definitions like LS_COLORS but I don't think you can count on it.

For ex. the Rust implementation of tldr, Tealdeer, uses a configuration file.
https://github.com/dbrgn/tealdeer/blob/main/docs/src/config.md
https://github.com/dbrgn/tealdeer/blob/main/docs/src/config_style.md

**Review by kbdharun — APPROVED** [review]

**Comment by EmilySeville7cfg** [comment]
image

Please also test my own CLIP mode. Example: `btldr-view sed`.
